ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -217,6 +236,36 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
     };
   }, [queryResults, thresholds, selectedLegendItems, legend, visual, isFetching, isLoading, y_axis?.max, y_axis?.min]);
 
```

Comment:
I'm not in love with putting another large chunk of code into `TimeSeriesChartPanel`, but I'm a little burned out on yak shaving to decompose this component at this point and really need to get the table legend up-and-running, so I'm doing it. Let me know if you feel strongly about not doing so.

Reply:
Sounds good, maybe let’s reconvene after we look more into the ‘time’ axis rewrite in #1134 too, that will also simplify a bit of the mapping, I’m planning to pair with Eunice on that soon to split up the TODOs